### ISSUE TITLE
socks5 support

### ISSUE THREAD

**Issue by mudler** [mudler/edgevpn#14]

**Comment by matti** [comment]
@mudler can you add a bit description here? Might give this a shot

**Comment by mudler** [comment]
Hi @matti ! thanks for showing interest in this.

The idea I had behind this feature is to provide a socket server from one side, and a "proxy" on the other side, similarly to https://github.com/mudler/edgevpn/blob/master/pkg/services/egress.go. 

Egress works that a node starts a proxy server and connections to the proxy are sent to another node which is advertised as egress (this needs to be explicitly enabled from the CLI) allows to accepts connection and proxy through them - the same thing could have be done with socks5, and hence this card!

I hope it was clear enough and spiked your interest :)

**Comment by matti** [comment]
What's the benefit of using socks5 instead of http proxy? lower overhead?

**Comment by mudler** [comment]
not really, just providing legacy support and isolation with old protocols. In this way we could be on pair with tor and provide the same abstraction layer
